app/dao/statistics_dao.py

```diff
@@ -0,0 +1,147 @@
+from datetime import datetime, timedelta
+from itertools import groupby
+
+from flask import current_app
+from sqlalchemy import func
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+
+from app import db
+from app.dao.dao_utils import transactional
+from app.models import (
+    JobStatistics,
+    Notification,
+    EMAIL_TYPE,
+    SMS_TYPE,
+    LETTER_TYPE,
+    NOTIFICATION_STATUS_TYPES_FAILED,
+    NOTIFICATION_DELIVERED,
+    NOTIFICATION_SENT)
+from app.statsd_decorators import statsd
+
+
+@transactional
+def timeout_job_counts(notifications_type, timeout_start):
+    total_updated = 0
+
+    sent = columns(notifications_type, 'sent')
+    delivered = columns(notifications_type, 'delivered')
+    failed = columns(notifications_type, 'failed')
+
+    results = db.session.query(
+        JobStatistics.job_id.label('job_id'),
+        func.count(Notification.status).label('count'),
+        Notification.status.label('status')
+    ).filter(
+        JobStatistics.job_id == Notification.job_id,
+        JobStatistics.created_at < timeout_start,
+        sent != failed + delivered
```

Comment:
yep done - doesn't change outcome but better.

(As columns in the stats table also do type restriction)